api/views/author.py

```diff
@@ -113,43 +53,32 @@ def post(self, request, *args, **kwargs):
         return Response(response_dict, status=status.HTTP_200_OK)
 
 
-class FollowerViewSet(
-    BaseRelationsMixin,
-    ModifyRelationsMixin,
-    viewsets.ViewSet
-):
+class FollowerViewSet(BaseRelationsMixin, viewsets.ViewSet):
     serializer_class = BaseRetrieveFollowingSerializer
+    follow_serializer_class = FollowRequestSerializer
 
-    # GET author/:pk (Returns a list of who you are following)
-    # GET author/:author_pk/follow/:pk (This follows the pk)
-    def retrieve(self, request, pk=None, author_pk=None):
-        """
-        Creating a following/follower relationship between authors
-
-        Takes:
-            pk: id of the person to be followed
-            author_pk: id of the person that will follow
-
-        If pk is none then we are retrieving the followers list
-        """
-        # Following the pk author
-        if author_pk:
-            author = get_object_or_404(self.queryset, id=author_pk)
-
-            # Can handle this in a permission
-            if request.user != author.user:
-                print "Trying to spoof author"
-                raise AuthenticationFailure
-
-            # Who we are following
-            follow = get_object_or_404(self.queryset, id=pk)
-            author.follow(follow)
-        else:
-            author = get_object_or_404(self.queryset, id=pk)
+    SAFE_METHODS = ['GET',]
 
+    # GET author/:author_pk/follow
+    # Returns who author_pk following
+    def list(self, request, author_pk=None):
```

Comment:
Lets not have this call at all. `GET /author/:id` returns the appropriate lists based on permission.